pylint/checkers/refactoring.py

```diff
@@ -261,13 +266,42 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
 
+        if self._check_iter_is_infer_list(node):
```

Comment:
Why are we doing this for lists? We should probably detect the rest of the mutable sequences, such as sets. But for the scope of this PR, we can leave it as is, let's just rephrase the name to ``_is_iterating_mutable_sequence``.